src/store/user.ts

```diff
@@ -22,17 +24,29 @@ export const useUserStore = defineStore('user', {
       privateKey: '',
       // This is just for initialization
       loginType: SocialLoginType.passwordless,
-      walletAddress: '',
+      ownerWalletAddress: '',
+      scwAddress: '',
       hasMfa: false,
       token: '',
     } as UserState),
   getters: {
     walletAddressShrinked(state: UserState): string {
-      const { walletAddress } = state
-      const walletAddressLength = walletAddress.length
-      return `${walletAddress.slice(0, 5)}....${walletAddress.slice(
-        walletAddressLength - 5
-      )}`
+      const rpcStore = store.state.value.rpcStore
+      const isGaslessConfigured =
+        rpcStore.gaslessEnabledStatus[rpcStore.selectedChainId] &&
+        rpcStore.preferredAddressType === 'scw'
+      const { ownerWalletAddress, scwAddress } = state
+      const address = isGaslessConfigured ? scwAddress : ownerWalletAddress
+      const addressLength = address.length
+      return `${address.slice(0, 5)}....${address.slice(addressLength - 5)}`
```

Comment:
You can use the `truncate` utility that we have

Reply:
[done](https://github.com/arcana-network/wallet-ui/pull/466/commits/8dcf923f3e831e4bacae4d1392d1c9233711408a)